File: C/postorder.c
#include <stdio.h> 
#include <stdlib.h> 
  
/* A binary tree node has data, pointer to left child 
   and a pointer to right child */
struct node 
{ 
     int data; 
     struct node* left; 
     struct node* right; 
}; 
  
/* Helper function that allocates a new node with the 
   given data and NULL left and right pointers. */
struct node* newNode(int data) 
{ 
     struct node* node = (struct node*) malloc(sizeof(struct node)); 
     node->data = data; 
     node->left = NULL; 
     node->right = NULL; 
  
     return(node); 
} 
  
/* Given a binary tree, print its nodes according to the 
  "bottom-up" postorder traversal. */
void printPostorder(struct node* node) 
{ 
     if (node == NULL) 
        return; 
  
     // first recur on left subtree 
     printPostorder(node->left); 
  
     // then recur on right subtree 
     printPostorder(node->right); 
  
     // now deal with the node 
     printf("%d ", node->data); 
} 
 
/* Driver program to test above functions*/
int main() 
{ 
     struct node *root  = newNode(1); 
     root->left             = newNode(2); 
     root->right           = newNode(3); 
     root->left->left     = newNode(4); 
     root->left->right   = newNode(5);  
   
     printf("\nPostorder traversal of binary tree is \n"); 
     printPostorder(root); 
  
     getchar(); 
     return 0; 
} 
/*
  Time Complexity - O(N)
  Space Complexity - O(N)
  
  Input - Its Taken in the Code .
  Output -Postrder traversal of binary tree is
          4 5 2 3 1
*/

File: JavaScript/postOrder.js
class Node {
    constructor(val) {
        this.val = val;
        this.left = null;
        this.right = null;
    }
}

class Tree {
    constructor() {
        this.root = null;
    }

    insert = val => {
        let root = new Node(val);
        return root;
    }

    postOrder = root => {
        if (root) {
            this.postOrder(root.left);
            this.postOrder(root.right);
            console.log(root.val);
        }
    }
}

const tree = new Tree();
let root = tree.root;
root = tree.insert(10);
root.left = tree.insert(4);
root.right = tree.insert(14);
root.right.left = tree.insert(12);
root.right.left.left = tree.insert(11);
root.left.right = tree.insert(7);
root.left.right.left = tree.insert(6);
root.left.right.right = tree.insert(8);
root.left.left = tree.insert(1);
root.left.right.right.right = tree.insert(9);

tree.postOrder(root);

/*
*  Example
*
*                10
*         4             14
*       1   7         12
*         6   8     11
*               9 
*
* for above tree, postorder is 1 6 9 8 7 4 11 12 14 10
*/